db/products.js

```diff
@@ -0,0 +1,82 @@
+const productCollection = {
+  products: [],
+};
+let productId = 1;
+
+function create(obj) {
+  const hasKeys = obj.name && obj.price && obj.inventory;
+
+  if (hasKeys) {
+    obj.id = productId;
+    productCollection.products.push(obj);
+    productId++;
+    return `{ "success": true}`;
+  } else if (!hasKeys) {
+    return `{ "success": false}`;
+  }
+}
+
+function retrieveAll() {
+  return productCollection;
+}
+
+function retrieveOne(productId) {
+  // check that productID is in collection.
+  const productIndex = productCollection.products.findIndex((elem) => elem.id === Number(productId));
+  if (productIndex === -1) {
+    return `{ "success": false. Product ID not found }`;
+  }
+
+  if (productIndex > -1) {
+    return productCollection.products[productIndex];
+  }
+}
+
+function update(obj, urlId) {
+  // check that productID is in collection.
+  const productIndex = productCollection.products.findIndex((elem) => elem.id === Number(urlId));
+  if (productIndex === -1) {
+    return `{ "success": false. Product ID not found }`;
+  }
+
+  // check that at least one product property is provided
+  const hasKeys = obj.name || obj.price || obj.inventory;
+  if (!hasKeys) {
+    return `{ "success": false. Provide field value(s) to modify. }`;
+  }
+
+  if (hasKeys && productIndex > -1) {
```

Comment:
Technically you don't need this check at all since both of these conditions have already been evaluated to be true in prior if conditions.